app/lib/feed_manager.rb

```diff
@@ -161,13 +161,19 @@ def filter_from_mentions?(status, receiver_id)
   # either action is appropriate.
   def add_to_feed(timeline_type, account, status)
     timeline_key = key(timeline_type, account.id)
-    reblog_key = key(timeline_type, account.id, 'reblogs')
+    reblog_key   = key(timeline_type, account.id, 'reblogs')
 
     if status.reblog?
+      reblog_set_key = key(timeline_type, account.id, "reblogs:#{status.reblog_of_id}")
+
       # If the original status or a reblog of it is within
       # REBLOG_FALLOFF statuses from the top, do not re-insert it into
       # the feed
       rank = redis.zrevrank(timeline_key, status.reblog_of_id)
+
+      redis.sadd(reblog_set_key, status.reblog_of_id) unless rank.nil?
+      redis.sadd(reblog_set_key, status.id)
```

Comment:
If you're concerned about Redis usage, we can alternatively do this only if another reblog is already within `REBLOG_FALLOFF`. It requires a bit more logic (to insert the other reblogging status, but only the first time, when creating the set), but doesn't increase Redis usage in the possibly-most-common case of only seeing one reblog.

Reply:
Mhhh I don't know what you mean

Reply:
If we want to minimize the stuff we add to Redis, we could:

* Not add to reblog_set_key for the *first* reblog.
* Only add to reblog_set_key for the second-or-later reblog.

(I had thought that we'd have to add the original reblog to reblog_set_key, but I actually don't think we would: if it gets deleted, our first move would be to remove it from reblog_set_key anyway. If we do want to track all of the reblogs for a status that a user could see (which would be nice to expose in the UI, but not what this PR appears to be about), then we'd have to add to that set, but for the limited purposes of this PR, we could get away with not creating the extra set in cases where there was only one user-visible reblog for a given status.)